src/docfx/build/Build.cs

```diff
@@ -108,6 +109,8 @@ private static void Run(Context context)
             // TODO: decouple files and dependencies from legacy.
             var dependencyMap = context.DependencyMapBuilder.Build();
 
+            MemoryCache.Clear();
```

Comment:
This could fix the issue by releasing the memory manually. 
But shall we rethink about the life cycle of the instances within `Context`?
Does everything in this "IoC container" need to keep the state all the time? 
For a real IoC container, we could configure the life cycle of the instance type.

Reply:
We will break down the dependencies based on precise usage, that would be further refactoring